src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -96,6 +96,9 @@ private sealed partial class Emitter
 
             private readonly HashSet<string> _emittedPropertyFileNames = new();
 
+            private bool _generateGetConverterMethodForTypes = false;
```

Comment:
nit: at first it wasn't clear to me that these were tied to the _currentContext and that _currentContext would actually change as the emitter did its work.  It might make sense to associate this mutable state to a single object or name it in a way that makes it clear how it applies and when it changes.

Reply:
For example as members of ContextGenerationSpec